diff --git a/devices/generic.c b/devices/generic.c
index f6cef9b5..38e7c331 100644
--- a/devices/generic.c
+++ b/devices/generic.c
@@ -234,7 +234,7 @@ int ec_gen_device_create_socket(
     sa.sll_family = AF_PACKET;
     sa.sll_protocol = htons(ETH_P_ETHERCAT);
     sa.sll_ifindex = desc->ifindex;
-    ret = kernel_bind(dev->socket, (struct sockaddr *) &sa, sizeof(sa));
+    ret = kernel_bind(dev->socket, (struct sockaddr_unsized *) &sa, sizeof(sa));
     if (ret) {
         printk(KERN_ERR PFX "Failed to bind() socket to interface"
                 " (ret = %i).\n", ret);
