--- a/vendor/Makefile	1970-01-01 01:00:01.000000000 +0100
+++ b/vendor/Makefile	2020-06-04 00:17:07.860421685 +0100
@@ -36,15 +36,11 @@
 
 include $(TOP)/configure/RULES
 
-$(TOP)/vendor/ffmpeg-%/ffbuild/config.mak: $(TOP)/vendor/yasm-%/yasm
+$(TOP)/vendor/ffmpeg-%/ffbuild/config.mak:
 	mkdir -p $(TOP)/vendor/ffmpeg-$(*F)
 	cd $(TOP)/vendor/ffmpeg-$(*F) && export PATH=${PATH}:../yasm-$(*F) && sh ../ffmpeg/configure $(CONFIGOPTIONS) && make && make install
 	$(MAKE) install
 
-$(TOP)/vendor/yasm-%/yasm:
-	mkdir -p $(TOP)/vendor/yasm-$(*F)
-	cd $(TOP)/vendor/yasm-$(*F) && sh ../yasm/configure && make
-	touch $@
 	
 # unfortunately the libs are not compatible with the opt/ref flag and pre-2012
 # MSVC compilers, so we'll have to make our own...
