diff --git a/Makefile b/Makefile
index 2db402a..ddf805c 100644
--- a/Makefile
+++ b/Makefile
@@ -72,7 +72,7 @@ SHARE_INSTDIR=$(DESTDIR)$(PREFIX)/share/minipro
 INCLUDE_INSTDIR=$(DESTDIR)$(PREFIX)/include/libminipro
 MAN_INSTDIR=$(DESTDIR)$(PREFIX)/share/man/man1
 
-UDEV_DIR=$(shell $(PKG_CONFIG) --define-variable=prefix=$(PREFIX) --silence-errors --variable=udevdir udev)
+UDEV_DIR ?= $(shell $(PKG_CONFIG) --define-variable=prefix=$(PREFIX) --silence-errors --variable=udevdir udev)
 UDEV_RULES_INSTDIR=$(DESTDIR)$(UDEV_DIR)/rules.d
 
 COMPLETIONS_DIR=$(shell $(PKG_CONFIG) --define-variable=prefix=$(PREFIX) --silence-errors --variable=completionsdir bash-completion)
